lib/io/can.hpp

```diff
@@ -1,8 +1,22 @@
+#include <string>
+
+#include <linux/can.h>
+
+#include <core/logger.hpp>
 #include <core/types.hpp>
 
 namespace hyped::io {
+
+enum CanResult { kFailure, kSuccess };
 class Can {
  public:
-  Can();
+  Can(hyped::core::ILogger &logger);
+  CanResult initialiseCanSocket(std::string can_network_interface);
+  CanResult sendCanFrame(const core::CanFrame message);
+  std::optional<can_frame> receiveCanFrame();
```

Comment:
why `can_frame` rather than `core::CanFrame`?

Reply:
I think we should stick with one of the options wherever we can. I believe that `can_frame` is not Darwin friendly so we should define `core::CanFrame` to be `can_frame` on Linux and custom on Darwin. Once we have done that, we only ever refer to `core::CanFrame` again, regardless of architecture.